src/nodedef.cpp

```diff
@@ -676,12 +676,12 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 
 	bool is_liquid = false;
 
-	u8 material_type = (alpha == 255) ?
-		TILE_MATERIAL_BASIC : TILE_MATERIAL_ALPHA;
```

Comment:
iIIRC it did not work with those lines unchanged, however, I agree that the behaviour could be preserved more but it will likely need some re-structuring.